python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -296,18 +317,25 @@ def measure(self, measRecord: lsst.afw.table.SourceRecord, exposure: afwImage.Ex
                 errorCollection[str(sF)] = error
                 continue
 
-            for sigma in self.config.sigmas:
+            for sigma in sigmas:
                 baseName = self.ConfigClass.getGaapResultName(sF, sigma, self.name)
-                if targetSigma >= sigma:
+                try:
+                    aperSigma2 = sigma**2 - targetSigma**2
+                    aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0, normalize=True)
+                    fluxScaling = 0.5*sigma**2/aperSigma2  # Eq. A16 of Kuijken et al. (2015)
+                except TypeError:
```

Comment:
You generally shouldn't use exceptions for control flow like this, especially for valid input. I don't see why this should be here instead of a `if sigma == 'PsfFlux':` block.

Reply:
The coding style in Python is to prefer try/except over if/else blocks, since the former is usually more efficient. The `try` block here is the most common case in normal mode of operation and avoids branching if the plugin is run with `doPsfPhotometry` is set to False.

Reply:
I see your point. I think the distinction is that I would try to avoid triggering exceptions that are inevitable with valid user input (config settings here). If you're going to refactor this anyway then the amended solution is fine.